src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,159 @@
+import React, { useEffect, useRef, useState } from 'react';
+import cn from 'classnames';
+import { Todo } from '../../types/Todo';
+import { UpdateTodoArgs } from '../../types/UpdateTodoArgs';
+
+interface Props {
+  todo: Todo;
+  deleteTodo: (todoId: number) => void;
+  toggleTodoStatus:(
+    todoId: number,
+    args: UpdateTodoArgs
+  ) => void;
+  updatingTodosId: number[]
+  selectedTodoId: number | null;
+  setSelectedTodoId: (todoId: number | null) => void;
+  updateTodoTitle: (todoId: number, args: UpdateTodoArgs) => void;
+}
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  deleteTodo,
+  toggleTodoStatus,
+  updatingTodosId,
+  selectedTodoId,
+  setSelectedTodoId,
+  updateTodoTitle,
+
+}) => {
+  const [editedTitle, setEditedTitle] = useState(todo.title);
+
+  const isUpdatingItem = updatingTodosId.includes(todo.id);
+
+  const handleToggleTodoStatus = () => toggleTodoStatus(
+    todo.id,
+    { completed: !todo.completed },
+  );
+
+  const clearSelectedTodoId = () => setSelectedTodoId(null);
+
+  const handleDoubleClick = (
+    event: React.MouseEvent<HTMLSpanElement, MouseEvent>,
+    todoId: number,
+  ) => {
+    event.preventDefault();
+    setSelectedTodoId(todoId);
+  };
```

Comment:
You are already inside your todo. You can use info from props

```suggestion
  const handleDoubleClick = (event: React.MouseEvent<HTMLSpanElement, MouseEvent>) => {
    event.preventDefault();
    
    setSelectedTodoId(todo.id);
  };
```